File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> numSet(nums.begin(), nums.end());

        int longest = 0;

        for(int num: numSet){
            if(!numSet.count(num-1)){
                int curr = num;
                int currentStreak = 1;

                while(numSet.count(curr + 1)){
                    curr++;
                    currentStreak++;
                }

                longest = max(longest,currentStreak);
            }
        }
        return longest;
    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        hash_map = {}

        for num in nums:
            hash_map[num] = hash_map.get(num,0) + 1
        
        duplicates = next((num for num, count in hash_map.items() if count > 1), None)
        if (duplicates != None):
            return True
        else:
            return False